backend-tests/tests/test_deployments.py

```diff
@@ -89,10 +89,13 @@ def create_tenant_test_setup(
             "POST", "/deployments", body=request_body
         )
         assert resp.status_code == 201
-    # Verify that the 'nr_deployments' expected deployments have been created
-    resp = api_mgmt_deploy.with_auth(user.utoken).call("GET", "/deployments")
-    assert resp.status_code == 200
-    assert len(resp.json()) == nr_deployments
+
+    if not isK8S():
+        # Verify that the 'nr_deployments' expected deployments have been created
+        resp = api_mgmt_deploy.with_auth(user.utoken).call("GET", "/deployments")
+        assert resp.status_code == 200
+        assert len(resp.json()) == nr_deployments
```

Comment:
Just to understand: would it be possible to get the "count" of deployments before running the test, then getting the "count" after it, and test if the difference is `== nr_deployments`. This applies to all asserts that are now conditional if `not isK8S()`.

Reply:
In case of deployments, it should be possible, because the endpoint for listing deployments returns `X-Total-Count`.
The endpoint for listing devices in the deviceauth doesn't have this functionality, but there is a separate endpoint for counting devices, so we can use it instead. I'll check rest of the cases.
There are places where we're not only checking number of the results, but also going through all the results. I'd still skip those cases when using existing tenant.

Reply:
done where it was enough to count documents before and after the test